### PR TITLE
Fix error handling when getting terms of service.

### PR DESCRIPTION
**Pull request by frankban**
Also make relative date tests work on all timezones.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5391/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5392/
Test FAILed.

**Comment by frankban** [comment]
This seems a spurious npm error, retrying.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5393/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5394/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5395/
Test FAILed.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 1]
```diff
@@ -115,10 +115,9 @@ YUI.add('user-profile', function() {
     */
     _getAgreementsCallback: function(error, data) {
       this.setState({loadingAgreements: false});
-      // We need to coerce error types returned bt the terms API into one error.
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: I'm guessing we no longer need this sort of coercion?

**Comment by kadams54** [comment]
LGTM. :+1: Thanks!

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: AFAIK this was never required.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5396/
Test PASSed.

**Comment by frankban** [comment]
Thanks for the review Kyle!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**